docs/django-migrations-standards.md

```diff
@@ -0,0 +1,29 @@
+# Django Migrations Standards
+
+From Django 3.2 documentation on [Migrations](https://docs.djangoproject.com/en/3.2/topics/migrations/#module-django.db.migrations)
+
+> Migrations are Django’s way of propagating changes you make to your models (adding a field, deleting a model, etc.) into your database schema. They’re designed to be mostly automatic, but you’ll need to know when to make migrations, when to run them, and the common problems you might run into.
+
+For creating migrations use Django's built-in commands [here](https://docs.djangoproject.com/en/3.2/topics/migrations/#the-commands).
+
+### Adding New Model Field and Migrations
+
+When adding new fields in an existing model, there is a possibility that the code referencing the fields can run before the new fields are populated with default values in the database. There are two ways to handle this race condition:
+
+1. Run `./manage.py sqlmigrate` and manually update the SQL query to accept `NULL` value for the new field in the databse. See example of this in [emails/migrations/0054_profile_forwarded_first_reply.py](https://github.com/mozilla/fx-private-relay/blob/main/emails/migrations/0054_profile_forwarded_first_reply.py). The migration for the newly added Django field `forwarded_first_reply` is set to default to `False` for new entries, while [this code](https://github.com/mozilla/fx-private-relay/blob/main/emails/migrations/0054_profile_forwarded_first_reply.py#L43) sets the database column `forwarded_first_reply` to allow `NULL` value. As a result, new instance of the model `Profile` will be populated with the defualt value `False` on the Django level while the old entries and the database level allows `forwarded_first_reply` column to be `NULL`.
```

Comment:
Almost everything is wrong with this paragraph...

"SQL Query" is for getting data out of the database. Commands to alter the schema are SQL statements.

The schema migrations statement will already accept `NULL` if the Django model has `null=True`. The issue is that Django does not support database-level defaults (Django 5.0 will have this feature). Django only uses defaults when _it_ writes to the database.


So, the change in `0054_profile_forwarded_first_reply.py` was to manually add a database-level default. This is easy in PostgreSQL:

```sql
ALTER TABLE "emails_profile"' ALTER COLUMN "forwarded_first_reply" SET DEFAULT false;
```

SQLite does not support [ALTER TABLE ... ALTER COLUMN](https://www.sqlite.org/lang_altertable.html), so the process is to _recreate_ the table with the default value:

```sql
CREATE TABLE "new__emails_profile" (
  ...
  ' "forwarded_first_reply" bool NULL SET DEFAULT false
); 
INSERT INTO "new__emails_profile" (
   ... _old_columns_ ...,
   "forwarded_first_reply" )
SELECT (
   ... _old_columns_ ...,
   false
) FROM emails_profile;
DROP TABLE "emails_profile";
ALTER TABLE "new__emails_profile" RENAME TO "emails_profile";
```

**This means `0054_profile_forwarded_first_reply.py` is wrong**, because it is missing `SET DEFAULT false` in the SQLite3 version.

Reply:
Moving this work to MPP-3464 on [2f945c1](https://github.com/mozilla/fx-private-relay/pull/3930/commits/2f945c1f9a2ac3a13bf60511efa391d49702e318)

Reply:
This comment is [referenced](https://mozilla-hub.atlassian.net/browse/MPP-3464?focusedCommentId=774050) in the MPP-3464